### PR TITLE
🌱 Added for import aliases for kubernetes packages

### LINKED ISSUE #1246: Make list of Go imports that we use aliases for and update golangci-lint
/kind feature

**Describe the solution you'd like**
Update golangci-lint with an extensive list of import aliases that we use, mainly from Kubernetes packages. Maybe take other repositories (Cluster API, Syself repos) as base.

Reference: https://github.com/syself/cluster-api-provider-hetzner/blob/main/.golangci-suggestions.yaml#L84C1-L89C22

### PR DESCRIPTION
**Pull request by yrs147**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1246

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squash commits
- [ ] include documentation
- [ ] add unit tests

### CODE REVIEW

File: .golangci.yaml [hunk 1]
```diff
@@ -75,6 +75,20 @@ linters-settings:
...
+        alias: clusterv1
+      - pkg: sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1beta1
+        alias: bootstrapv1
+      - pkg: sigs.k8s.io/controller-runtime/pkg/metrics/server
+        alias: metricsserver        
+      - pkg: sigs.k8s.io/cluster-api/errors
+        alias: capierrors    
+      - pkg: sigs.k8s.io/controller-runtime/pkg/client/fake
+        alias: fakeclient       
+      - pkg: sigs.k8s.io/controller-runtime/pkg/client
```
Comment (janiskemper) [inline .golangci.yaml:90]: I am a bit hesitant with this one. I would remove it. "c" is just not a great choice for any alias, it is more a workaround :D
Did all the code comply to these aliases?

Comment (yrs147) [reply .golangci.yaml:90]: Yes it does , I went through it .
Then I'll remove this alias and push the changes

Comment (kranurag7) [reply .golangci.yaml:90]: > "c" is just not a great choice

How about using `ctrlc` given we already use `ctrl` for controller runtime.

Comment (janiskemper) [reply .golangci.yaml]: that is an idea. But at this point I wouldn't like to enforce this

**Review by janiskemper — DISMISSED** [review]

**Comment by janiskemper** [comment]
@kranurag7 this PR did not trigger any Go-related CI jobs.. It should of course.. This is about linting Go! Can you fix this?

**Comment by janiskemper** [comment]
converting to draft until checks are running locally

File: .golangci.yaml [hunk 2]
```diff
@@ -77,8 +77,6 @@ linters-settings:
         alias: ctrl
       - pkg: k8s.io/apimachinery/pkg/util/runtime
         alias: utilruntime
-      - pkg: sigs.k8s.io/cluster-api/api/v1beta1
```
Comment (janiskemper) [inline .golangci.yaml]: why did you remove this? Looks valid!

Comment (yrs147) [reply .golangci.yaml]: was throwing error in the CI, also idk why but I did run `make verify` before pushing the changes and it seems to be working fine but its failing in the CI

Comment (yrs147) [reply .golangci.yaml]: Ran into some issues , fixing it

Comment (kranurag7) [reply .golangci.yaml]: Please use the following: 
```bash
$ make lint-golang-ci
docker run  --rm -t -i \
	-v /home/ria/go/pkg:/go/pkg \
	-v /home/ria/caph:/src/cluster-api-provider-hetzner \
	ghcr.io/syself/caph-builder:1.0.16 lint-golang-ci;
make: Entering directory '/src/cluster-api-provider-hetzner'
go version
go version go1.21.5 linux/amd64
golangci-lint version
golangci-lint has version 1.57.2 built with go1.22.1 from 77a8601a on 2024-03-28T19:01:11Z
golangci-lint run --out-format=github-actions
::error file=api/v1beta1/hcloudmachine_types.go,line=23,col=2::import "sigs.k8s.io/cluster-api/errors" imported without alias but must be with alias "capierrors" according to config (importas)
make: *** [Makefile:699: lint-golang-ci] Error 1
make: Leaving directory '/src/cluster-api-provider-hetzner'
make: *** [Makefile:692: lint-golang-ci] Error 2
```

**Comment by janiskemper** [comment]
plz add the aliases again that you removed ;)

**Comment by yrs147** [comment]
> plz add the aliases again that you removed ;)

those are causing errors in the checks

**Review by janiskemper — COMMENTED** [review]
thanks! You can squash now!

**Comment by yrs147** [comment]
@janiskemper I've squashed the commits

**Comment by kranurag7** [comment]
@yrs147 please rebase as well because of settings.

**Review by janiskemper — APPROVED** [review]

**Review by kranurag7 — APPROVED** [review]
